Doc/reference/compound_stmts.rst

```diff
@@ -820,7 +820,9 @@ and binds no name.  Syntax:
 .. productionlist:: python-grammar
    wildcard_pattern: '_'
 
-``_`` is a :ref:`soft keyword <soft-keywords>`.
+``_`` is a :ref:`soft keyword <soft-keywords>` within any pattern,
+but only within patterns.  It is an identifier, as usual, even within
+``match`` headers, ``guards``, and ``case`` blocks.
```

Comment:
Terry, thanks for doing this. I think you're right - this is much clearer and users don't have to search through python grammar to see that ``capture_pattern`` only happens in ``patterns``.

On a side note: I think saying "``match`` subject expressions" rather than "``match`` headers" is more consistent with the rest of the terminology. I'll send a PR to update the language reference soon as there's a bunch of other stuff needing updating anyways.